clang/runtime/dpct-rt/include/dpct/sparse_utils.hpp

```diff
@@ -40,13 +40,119 @@ class matrix_info {
   oneapi::mkl::uplo _uplo = oneapi::mkl::uplo::upper;
   oneapi::mkl::index_base _index_base = oneapi::mkl::index_base::zero;
 };
+namespace detail {
+struct csrgemm_args_info {
+  csrgemm_args_info() {}
+  csrgemm_args_info(
+      oneapi::mkl::transpose trans_a, oneapi::mkl::transpose trans_b, int m,
```

Comment:
How about using tuple? Then we can get rid of the user-defined hash and operator==.

Reply:
tuple has == but not hash